src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -80,38 +55,86 @@ public void checkHasAccessToIrisSession(IrisSession session, User user) {
         if (user == null) {
             user = userRepository.getUserWithGroupsAndAuthorities();
         }
-        getIrisSessionSubService(session).checkHasAccessToIrisSession(session, user);
+        var wrapper = getIrisSessionSubService(session);
+        wrapper.irisSubFeatureInterface.checkHasAccessTo(user, wrapper.irisSession);
     }
 
     /**
      * Sends a request to Iris to get a message for the given session.
      * It decides which Iris subsystem should handle it based on the session type.
-     * Currently, only the chat subsystem exists.
      *
      * @param session The session to get a message for
      */
     public void requestMessageFromIris(IrisSession session) {
-        getIrisSessionSubService(session).requestAndHandleResponse(session);
+        var wrapper = getIrisSessionSubService(session);
+        if (wrapper.irisSubFeatureInterface instanceof IrisChatBasedFeatureInterface) {
+            ((IrisChatBasedFeatureInterface<?>) wrapper.irisSubFeatureInterface).requestAndHandleResponse(wrapper.irisSession);
+        }
```

Comment:
Can you directly name this object using instanceof pattern matching here instead of using a cast?

In general, this code looks quite strange (instanceoff on an attribute of a wrapper, generic wildcard). I'm not that familiar with Iris to suggest a better approach, but please check if this is really how you need to handle this case.

Reply:
Same for the whole file

Reply:
I agree with you both. This was pretty dumb and I noticed it today while coding a follow up Hestia PR.
I changed it now to use Generics in a more sensible manner.